core/shared/src/main/scala/zio/logging/LogFormat.scala

```diff
@@ -248,26 +246,36 @@ object LogFormat {
   /**
    * Returns a new log format that appends all annotations to the log output.
    */
-  def annotations: LogFormat =
+  def annotations: LogFormat = annotations(Set.empty)
+
+  def annotations(excludeKeys: Set[String]): LogFormat =
     LogFormat.make { (builder, _, _, _, _, _, _, _, annotations) =>
       annotations.foreach { case (key, value) =>
-        builder.appendKeyValue(key, value)
+        if (!excludeKeys.contains(key)) {
+          builder.appendKeyValue(key, value)
+        }
       }
     }
 
-  def logAnnotations: LogFormat =
+  def logAnnotations: LogFormat = logAnnotations(Set.empty)
+
+  def logAnnotations(excludeKeys: Set[String]): LogFormat =
```

Comment:
I would make `final case class ExcludeKeys(keys: Set[String])` just so we can at some point also have include keys and not break code.